routers/api/v1/utils/git.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package utils
```

Comment:
As lunny worried utils is becoming a bit of a catch-all package. Not necessarily this PR but I wonder if we these would be better in a git package.

Reply:
will create an issue, this is a bugfix not a whole refactor ... you can find more similar code (git ref -> sha lookup) all over different places and we should unify them

Reply:
-> #16411